### PR TITLE
Commons - exporting props

### PR DESCRIPTION
**Pull request by Inbal-Tish**
#863  Description
Exporting BaseComponentInjectedProps, ForwardRefInjectedProps and ModifiersProps (to extract sub-modifiers props, like MarginModifiers).

## Changelog
Commons - exporting props.

### CODE REVIEW

File: src/commons/index.js [hunk 1]
```diff
@@ -27,5 +27,14 @@ module.exports = {
   },
   get withScrollReached() {
     return require('./withScrollReached').default;
+  },
```
Comment (ethanshar) [inline src/commons/index.js:30]: This one is not good.. 
There's a new temp file `commons/new.ts` (that will replace commons/index.js eventually) 
The typings are already exported there..

File: src/index.js [hunk 2]
```diff
@@ -226,6 +226,15 @@ export default {
...
   },
+  get BaseComponentInjectedProps() {
+    return require('./commons').BaseComponentInjectedProps;
+  },
+  get ForwardRefInjectedProps() {
+    return require('./commons').ForwardRefInjectedProps;
+  },
+  get ModifiersProps() {
+    return require('./commons').default;
+  },
```
Comment (ethanshar) [inline src/index.js:237]: You need to export the typings in the ts files
`generatedTypes/index.d.ts`
`src/index.ts`

Comment (Inbal-Tish) [reply src/index.js:237]: Sorry, I don't understand what is what with all the fake files... Why they should be in `generatedTypes/index` if these are TS fils? And where all the other commons in `src/index.ts`?

File: src/commons/index.js (hunk 1 above)
Comment (Inbal-Tish) [reply src/commons/index.js:30]: Where is the call for the `commons/new.ts` file? I see only `commons/index.js`...

Comment (ethanshar) [reply src/commons/index.js:30]: We're using it in the components that need these typings.. 
View, Text, Button.. almost all of them..

**Pull request closed without merging**